### PR TITLE
Add succeed ✔, fail ✖ and stopAndPersist 🦄

### LINKED ISSUE #8: Add new methods .succeed() and .fail()?
Hey,
first of all, great module! Love it.

I thought that since this module is meant to display progress in something, wouldn't it make sense to also display whether that succeeds or fails?

What I'm imagining is 1) a method `.suceed()` that turns the spinner into a green tick, and 2) a method `.fail()` that turns the spinner into a red cross (and possibly the text, too?).

What does everyone think about this?

### PR DESCRIPTION
**Pull request by matheuss**
Closes #8.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -103,6 +103,20 @@ Start the spinner. Returns the instance.
...
 
+### .succeed()
+
+Stop the spinner, change it to a green `✔` and persists the `text`. See the gif below. Returns the instance.
+
+### .fail()
+
+Stop the spinner, change it to a red `✖` and persists the `text`. See the gif below. Returns the instance.
+
+### .stopAndPersist(icon)
```
Comment (sindresorhus) [inline readme.md]: `icon` => `symbol`

File: index.js [hunk 2]
```diff
@@ -84,6 +85,18 @@ class Ora {
 
 		return this;
 	}
+	succeed() {
+		return this.stopAndPersist(logSymbols.success);
+	}
+	fail() {
+		return this.stopAndPersist(logSymbols.error);
+	}
+	stopAndPersist(icon) {
```
Comment (sindresorhus) [inline index.js]: `icon` => `symbol`

File: readme.md (hunk 1 above)
Comment (sindresorhus) [inline readme.md]: I think we should make `symbol` optional and just print nothing there if not supplied.

**Comment by sindresorhus** [comment]
Looks great! :)

Can you it to example.js?

Something like:

```js
setTimeout(() => {
	spinner.succeed();
}, 2000);
```

**Comment by sindresorhus** [comment]
@SamVerschueren Can you make sure this works for you in Listr?

I'm not entirely sure how Listr works, but this calls `#stop()` which [tries to show the cursor](https://github.com/matheuss/ora/blob/0d4280aeb4ec83b207947ff84963b3c92489d08a/index.js#L84). That might cause some kind of cursor flicker for you if you start another instance right away.

File: readme.md (hunk 1 above)
Comment (matheuss) [inline readme.md]: Done ✨

File: index.js (hunk 2 above)
Comment (matheuss) [inline index.js]: Done 🔥

**Comment by matheuss** [comment]
Thanks for the feedback Sindre 😊

I don't that mgiht be a problem to `Listr`, given that this code:

```js
spinner1.start();
setTimeout(() => {
	spinner1.succeed();
	spinner2.start();
}, 1000);

setTimeout(() => {
	spinner2.fail();
	spinner3.start();
}, 2000);

setTimeout(() => {
	spinner3.stopAndPersist('🦄');
}, 3000);
```

Produces this:
image

**Comment by niftylettuce** [comment]
bump

**Comment by sindresorhus** [comment]
Looks great @matheuss. Thank you :)

And sorry about the delay. I've been AFK for some days.

**Comment by sindresorhus** [comment]
Changelog: https://github.com/sindresorhus/ora/releases/tag/v0.3.0

**Comment by sindresorhus** [comment]
@matheuss Invited you to the repo since this is an [open open source](http://openopensource.org) project :)

If you're not interested, just decline. No worries.

**Comment by matheuss** [comment]
I'm more than interested! ☺️ Thanks!

**Pull request merged**